edgedb/datatypes/namedtuple.c

```diff
@@ -220,11 +237,88 @@ namedtuple_new(PyTypeObject *type, PyObject *args, PyObject *kwargs) {
 
 
 static PyObject *
-namedtuple_getattr(EdgeNamedTupleObject *o, PyObject *name)
+namedtuple_derived_new(PyTypeObject *type, PyObject *args, PyObject *kwargs) {
```

Comment:
What is this function used for?

Will it be used with `a_fetched_named_tuple.__class__((1, 2, 3))`?

Reply:
Yes, in `dataclasses.py`:

```python
    elif isinstance(obj, tuple) and hasattr(obj, '_fields'):
        return type(obj)(*[_asdict_inner(v, dict_factory) for v in obj])
```